### PR TITLE
Fix/syntax check

### PR DESCRIPTION
**Pull request by Kevinjil**
Currently, only files in the `.github/jobs` directory are tested.

### CODE REVIEW

**Review by nickygerritsen — APPROVED** [review]

File: webapp/src/Controller/API/ContestController.php
Comment (meisterT) [inline webapp/src/Controller/API/ContestController.php:1]: Why the changes to this file?

Comment (Kevinjil) [reply webapp/src/Controller/API/ContestController.php:1]: These were test changes, which should not have been pushed. Hence the draft and force-push.

File: .github/jobs/syntax-check [hunk 1]
```diff
@@ -3,27 +3,29 @@
...
 if [ ! -x /usr/bin/shellcheck ]; then
 	echo "/usr/bin/shellcheck not found or not executable"
 	# shellcheck disable=SC2242
-	exit -1
+	exit 1
 fi
 
 find . \( \
-       -path ./webapp/vendor        -prune \
+       -path ./webapp/vendor     -prune \
```
Comment (vmcj) [inline .github/jobs/syntax-check:28]: I suspect there is a tab here for all lines.

File: judge/runguard_test/runguard_test.sh [hunk 2]
```diff
@@ -156,14 +159,15 @@ test_redir_stderr() {
...
+	actual=$(wc -c < "$stderr")
 	[ $limit -eq $actual ] || fail "redirected stdout not limited to ${limit}B, but wrote ${actual}B"
 
 	rm "$stderr"
 }
 
 test_rootdir_changedir() {
 	# Prepare test directory.
+	# shellcheck disable=SC2154
 	almost_empty_dir="$judgehost_judgedir/runguard_tests/almost_empty"
```
Comment (vmcj) [inline judge/runguard_test/runguard_test.sh:171]:
```suggestion
	almost_empty_dir="${judgehost_judgedir}/runguard_tests/almost_empty"
```

File: judge/version_check.sh [hunk 3]
```diff
@@ -66,6 +66,7 @@ touch version_check.out version_check.meta
 if [ -e "$WORKDIR/version_check-script" ]; then
     mv "$WORKDIR/version_check-script" "$WORKDIR/version_check-script-old"
 fi
+# shellcheck disable=SC2174
 mkdir -m 0777 -p "$WORKDIR/version_check-script"
```
Comment (vmcj) [inline judge/version_check.sh:70]: I think making the behaviour explicit is better see: https://www.shellcheck.net/wiki/SC2174

File: sql/files/defaultdata/r/run [hunk 4]
```diff
@@ -30,10 +30,14 @@ fi
 #
 # Store intermediate files in the current dir (/compile) as its only
 # available during compilation step.
-export TMPDIR=`pwd`
-Rscript -e "parse('"$@"')"
-EXITCODE=$?
-[ "$EXITCODE" -ne 0 ] && exit $EXITCODE
+TMPDIR=`pwd`
+export TMPDIR
+for f in "$@"
```
Comment (vmcj) [inline sql/files/defaultdata/r/run:35]: Does this need to be quoted?

File: sql/dj_setup_database.in [hunk 5]
```diff
@@ -407,19 +406,20 @@ load)
...
 		ind=1
-		for i in "$databases"; do
+		for i in $databases; do
 			echo "$ind) $i"
-			: $((ind+=1))
+			ind=$((ind+1))
 		done
 		while true; do
-			read -p "Which database should be loaded? " db
+			echo "Which database should be loaded? "
```
Comment (vmcj) [inline sql/dj_setup_database.in:414]: Doesn't the `printf` not fix the mentioned issue?

File: sql/files/defaultdata/r/run (hunk 4 above)
Comment (Kevinjil) [reply sql/files/defaultdata/r/run:35]: https://www.shellcheck.net/wiki/SC2068

File: judge/runguard_test/runguard_test.sh (hunk 2 above)
Comment (Kevinjil) [reply judge/runguard_test/runguard_test.sh:171]: Does not fix SC2154

File: .github/jobs/syntax-check (hunk 1 above)
Comment (Kevinjil) [reply .github/jobs/syntax-check:28]: This actually fixes the wrong indentation

**Review by vmcj — APPROVED** [review]
I can live with this, let's merge and actually have a test again.
